acvm/src/pwg/memory_op.rs

```diff
@@ -97,9 +110,19 @@ impl MemoryOpSolver {
             // into the memory block.
             let value_write = value;
 
-            let value_to_write = get_value(&value_write, initial_witness)?;
-
-            self.write_memory_index(memory_index, value_to_write)
+            // A zero predicate indicates that we should skip the write operation and zero out
+            // the memory it touches.
```

Comment:
I also debated just not performing a write here at all, but was unsure on which was the safer approach